neonvm/controllers/metrics.go

```diff
@@ -121,11 +154,22 @@ func (d *wrappedReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	// async approach.
 	d.lock.Lock()
 	defer d.lock.Unlock()
+	outcome := SuccessOutcome
 	if err != nil {
 		d.failing[req.NamespacedName] = struct{}{}
```

Comment:
WDYT about using separate maps / counts for failure vs conflict?

Reply:
Created a follow-up https://github.com/neondatabase/autoscaling/issues/932